docs/src/components/CustomMediaStreamControl.tsx

```diff
@@ -0,0 +1,59 @@
+import 'webrtc-adapter';
+import * as React from 'react';
+import { IMediaStreamControlHandle } from 'react-conf-webrtc';
+
+export interface ICustomMediaStreamControlProps {
+    handle: IMediaStreamControlHandle;
+}
+
+export interface ICustomMediaStreamControlStates {
+    isAudioEnabled: boolean;
+    isVideoEnabled: boolean
+}
+
+export default class CustomMediaStreamControl extends React.Component<ICustomMediaStreamControlProps, ICustomMediaStreamControlStates> {
+    constructor(props: ICustomMediaStreamControlProps) {
+        super(props);
+
+        const { handle } = props;
+
+        this.state = {
```

Comment:
Don't use state for this, because it creates two sources of truth. Use props instead.

Reply:
+1

Reply:
In the sense, `isAudioEnabled` or `isVideoEnabled` changed will trigger `CustomMediaStreamControl` component re-render and update the button text(eg: Disable Video to Enable Video)

When we call setAudio(Video)Enabled, the component will not trigger re-render directly.

This is why I define state here.

Reply:
The renderer does not need to trigger re-render. It will re-render whenever the props change.

Reply:
Are you saying the SetAudio(Video) is not mutating the object in the props?

Reply:
@asinner 

> let the MediaStreamControl trigger the re-render

Use a ref and call CustomMediaStreamControlRef.forceUpdate() ?

Reply:
@linweigao stream.audioTrack is a `side effect` here. 
When we change the audioTrack.enabled value, react don't trigger re-render. 
So, we needs to store the enabled value to react's state or props